app/scripts/views/settings/clients.js

```diff
@@ -130,16 +130,17 @@ define(function (require, exports, module) {
 
     _onDisconnectClient: function (event) {
       var item = this._attachedClients.get($(event.currentTarget).data('id'));
-
-      return this.user.destroyAccountClient(this.user.getSignedInAccount(), item)
-        .then(() => {
-          var clientType = item.get('clientType');
-
-          this.logViewEvent(clientType + '.disconnect');
-          if (clientType === Constants.CLIENT_TYPE_DEVICE && item.get('isCurrentDevice')) {
-            this.navigateToSignIn();
-          }
+      var clientType = item.get('clientType');
+      this.logViewEvent(clientType + '.disconnect');
+      // if a device then ask for confirmation
+      if (clientType === Constants.CLIENT_TYPE_DEVICE) {
+        this.navigate('settings/clients/disconnect', {
+          clients: this._attachedClients,
```

Comment:
Yeah I had it working that away initially, but client must be destroyed from the `clients` collection. Otherwise it will be calling Backbone sync on destroy and XHR our: http://backbonejs.org/#Model-destroy and also this helps with:  https://github.com/mozilla/fxa/pull/181/files#diff-9d765e5787806afc80bdaa86191886a8R199 ( After device removal has been confirmed, the dialog closes and, the row is overtaken by the area beneath it)